kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java

```diff
@@ -107,6 +119,70 @@ public Row getScanState(TableClient tableClient) {
 
     @Override
     public Optional<Predicate> getRemainingFilter() {
-        return filter;
+        return getDataFilters();
+    }
+
+    private Optional<Tuple2<Predicate, Predicate>> splitFilters(Optional<Predicate> filter) {
+        return filter.map(predicate -> {
+            Metadata metadata = protocolAndMetadata.get()._2;
+            Set<String> partColNames = metadata.getPartitionColumns().stream().collect(toSet());
+            return PartitionUtils.splitMetadataAndDataPredicates(predicate, partColNames);
+        });
+    }
+
+    private Optional<Predicate> getDataFilters() {
+        return removeAlwaysTrue(partitionAndDataFilters.get().map(filters -> filters._2));
+    }
+
+    private Optional<Predicate> getPartitionsFilters() {
+        return removeAlwaysTrue(partitionAndDataFilters.get().map(filters -> filters._1));
+    }
+
+    /**
+     * Consider `ALWAYS_TRUE` as no predicate.
+     */
+    private Optional<Predicate> removeAlwaysTrue(Optional<Predicate> predicate) {
+        return predicate.map(filter -> {
+            if (filter.getName().equalsIgnoreCase("ALWAYS_TRUE")) {
+                return null;
```

Comment:
hm. how does returning null actually filter anything?

instead of `predicate.map` shouldn't we just do `predicate.filter`?

im sure youre right, i just don't understand. perhaps add a comment explaining how `null` is consumed downstream and how it effectively removes the filter.